gem2rpm/foreman_plugin.spec.erb

```diff
@@ -56,27 +56,21 @@ BuildRequires: <%= requirement "ruby#{'-devel' unless spec.extensions.empty?}",
 <% for req in spec.required_rubygems_version -%>
 BuildRequires: <%= requirement 'rubygems-devel', req %>
 <% end -%>
-<% unless spec.extensions.empty? -%>
+<% if !spec.extensions.empty? || has_assets || has_webpack -%>
 <% for dep in runtime_dependencies.virtualize -%>
-<% for req in dep.requirement -%>
-BuildRequires: <%= requirement(dep.name, req) %>
-<% end -%>
+<% build_requires = dep.requirement.map { |req| requirement(dep.name, req) } -%>
+BuildRequires: <%= build_requires.size > 1 ? "(#{build_requires.join(' with ')})" : build_requires.first %>
 <% end -%>
+<% unless spec.extensions.empty? -%>
 # Compiler is required for build of gem binary extension.
 # https://fedoraproject.org/wiki/Packaging:C_and_C++#BuildRequires_and_Requires
 BuildRequires: gcc
 <% end -%>
+<% end -%>
 <% if spec.extensions.empty? -%>
 BuildArch: noarch
 <% end -%>
 Provides: foreman-plugin-%{plugin_name} = %{version}
```

Comment:
Given https://github.com/theforeman/foreman-packaging/pull/10024#discussion_r1404559804, I think there is something wrong with the if/end here, but I don't see what.

Reply:
I can move it up for a smaller diff. And I can simplify the conditions.